appcli/stack_manager/archive_manager.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python3
+# # -*- coding: utf-8 -*-
+
+"""
+Handles archiving and rotation of logs and files.
+________________________________________________________________________________
+
+Created by brightSPARK Labs
+www.brightsparklabs.com
+"""
+
+
+# -----------------------------------------------------------------------------
+# IMPORTS
+# -----------------------------------------------------------------------------
+
+# Standard imports.
+from typing import Literal, List, Union
+from pathlib import Path
+import datetime
+import tarfile
+import glob
+
+# Vendor imports.
+from pydantic import BaseModel
+
+# Local imports.
+from appcli.models.cli_context import CliContext
+from appcli.logger import logger
+
+
+# -----------------------------------------------------------------------------
+# CLASSES
+# -----------------------------------------------------------------------------
+
+
+class CompressRule(BaseModel):
+    """Rule to compress files into an archived file.
+    This will overwrite any archive file that already exists.
+    """
+
+    type: Literal["compress"] = "compress"
+    """Explicitly declare rule type."""
+
+    name: str
+    """Name of the rule."""
+
+    include_list: List[str] = ["[]"]
+    """Glob pattern of files/dirs in `data_dir` to include in the archive."""
+
+    archive_file: str
+    """Archive file in `data_dir` to compress the files into.
+    Dateime is supported using the `%<value>` notation, i.e:
```

Comment:
typo Dateime

We should be defaulting this to something like:

```
%Y-%m%d_myapp_<shell-safe/sanitised name>.tgz
```

I think that's more aligned with our actual backup file names. We should be consistent with that naming style.